crd-generator/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java

```diff
@@ -183,7 +185,18 @@ public T internalFrom(TypeRef typeRef) {
         if (typeRef instanceof ClassRef) { // Handle complex types
           ClassRef classRef = (ClassRef) typeRef;
           TypeDef def = classRef.getDefinition();
-          return internalFrom(def);
+
+          // check if we're dealing with an enum
+          if(def.isEnum()) {
+            final JsonNode[] enumValues = def.getProperties().stream()
+              .map(Property::getName)
+              .map(JsonNodeFactory.instance::textNode)
```

Comment:
It seems really weird to me that we have to use Jacson from within our mapping mechanism.
Then again this is how our `JsonSchemaProps` class is structured and we can't do much about it. At least not while we are trying to deal with the enum issue.